### ISSUE TITLE
alignof specifier

### ISSUE THREAD

**Issue by Cazadorro** [microsoft/DirectXShaderCompiler#4595]
With https://github.com/microsoft/DirectXShaderCompiler/pull/4226 knowing alignment has become even more important.  Alignment should be known at compile time from said type.  It would be great to have [alignof](https://en.cppreference.com/w/cpp/language/alignof) specifier like both C and C++ have, allowing wrappers to eliminate alignment from vk::RawBufferLoad calls. 

I attempted to create this function myself with current utilities, but const expression propagation does not appear to be respected through functions, so it's not possible to simply to use overloading or template specialization to create such a function manually.

**Comment by devshgraphicsprogramming** [comment]
yes that would be awesome, and I'd also like to have a `static_assert` or runtime assert in the `RawBufferLoad` when you specify an alignment of less that `alignof(T)`
